caravel/assets/javascripts/explorev2/components/charts/TimeSeriesLineChart.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { PropTypes } from 'react';
+import * as V from 'victory';
+import theme from '../../../components/VictoryTheme';
+import moment from 'moment';
+import { schemeCategory20c } from 'd3-scale';
+import Legend from './Legend';
+
+const propTypes = {
+  data: PropTypes.array.isRequired,
+  label1: PropTypes.string.isRequired,
+  height: PropTypes.number.isRequired,
+};
+
+export default class TimeSeriesLineChart extends React.Component {
+  constructor(props) {
+    super(props);
+    this.keysToColorsMap = this.mapKeysToColors(props.data);
```

Comment:
methods only need to be bound in certain cases, first if the method needs access to `this` and depending on where the function is getting called, ie.  in a callback after an event is triggered.